drivers/counter/rtc_mcp7940n.c

```diff
@@ -0,0 +1,785 @@
+/*
+ * Copyright (c) 2019-2020 Peter Bigot Consulting, LLC
+ * Copyright (c) 2021 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mcp7940n
+
+#include <device.h>
+#include <drivers/counter.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <drivers/rtc/mcp7940n.h>
+#include <kernel.h>
+#include <logging/log.h>
+#include <sys/timeutil.h>
+#include <sys/util.h>
+#include <time.h>
+
+LOG_MODULE_REGISTER(MCP7940N, CONFIG_COUNTER_LOG_LEVEL);
+
+/* Alarm channels */
+#define ALARM0_ID			0
+#define ALARM1_ID			1
+
+/* Size of block when writing whole struct */
+#define RTC_TIME_REGISTERS_SIZE		sizeof(struct mcp7940n_time_registers)
+#define RTC_ALARM_REGISTERS_SIZE	sizeof(struct mcp7940n_alarm_registers)
+
+/* Largest block size */
+#define MAX_WRITE_SIZE                  (RTC_TIME_REGISTERS_SIZE)
+
+/* tm struct uses years since 1900 but unix time uses years since
+ * 1970. MCP7940N default year is '1' so the offset is 69
+ */
+#define UNIX_YEAR_OFFSET		69
+
+/* Macro used to decode BCD to UNIX time to avoid potential copy and paste
+ * errors.
+ */
+#define RTC_BCD_DECODE(reg_prefix) (reg_prefix##_one + reg_prefix##_ten * 10)
+
+struct gpios {
+	const char *ctrl;
+	gpio_pin_t pin;
+	gpio_dt_flags_t flags;
+};
```

Comment:
Please use a gpio_dt_spec instead of reinventing it.

Reply:
Updated.